### ISSUE TITLE
`Send failure: Broken pipe` uploading 4.6MB crate

### ISSUE THREAD

**Issue by mmastrac** [kellnr/kellnr#180]
I'm not sure what's going on, but I'm able to publish smaller crates to a docker container but larger crates are failing.

Docker command to run. Note that bumping `KELLNR_REGISTRY__MAX_CRATE_SIZE` doesn't appear to help:

```
docker run --rm -it     -p 8000:8000  -e KELLNR_REGISTRY__MAX_CRATE_SIZE=20 -e KELLNR_LOG__LEVEL=trace -e KELLNR_LOG__LEVEL_WEB_SERVER=trace  ghcr.io/kellnr/kellnr:5.0.0
```

It looks like there's something in the web stack that's dropping larger requests on the floor, but not leaving any trace in the logs.

```
    Packaged 96 files, 11.2MiB (4.6MiB compressed)
   Uploading deno_core v0.250.0 (/Users/matt/Documents/github/deno/deno_core/core)
error: failed to publish to registry at http://127.0.0.1:8000

Caused by:
  [55] Failed sending data to the peer (Send failure: Broken pipe)
```

There are no obvious errors in the logs:

```
2024-01-24T17:47:26.602933Z DEBUG request: tower_http::trace::on_response: finished processing request latency=1 ms status=200 method=GET uri=/api/v1/crates/se/rd/serde_v8 version=HTTP/1.1
2024-01-24T17:47:26.604148Z DEBUG request: tower_http::trace::on_response: finished processing request latency=2 ms status=200 method=GET uri=/api/v1/crates/de/no/deno_ops version=HTTP/1.1
2024-01-24T17:47:27.891855Z DEBUG request: tower_http::trace::on_request: started processing request method=PUT uri=/api/v1/crates/new version=HTTP/1.1
2024-01-24T17:47:27.928979Z DEBUG request: tower_http::trace::on_response: finished processing request latency=37 ms status=200 method=PUT uri=/api/v1/crates/new version=HTTP/1.1
```

**Comment by mmastrac** [comment]
According to Wireshark, it dies approx 2784kB into the stream with this error:

```
HTTP/1.1 100 Continue

HTTP/1.1 200 OK
content-type: application/json
content-length: 66
date: Wed, 24 Jan 2024 17:57:21 GMT

{"errors":[{"detail":"ERROR: Failed to buffer the request body"}]}
```

**Comment by secana** [comment]
The max_crate_size was not respected. Fixed in the next Kellnr version 5.1.2 (comes in a few days)
